src/app/hearing-part/components/hearing-parts-preview/hearing-parts-preview.component.ts

```diff
@@ -15,49 +16,75 @@ export class HearingPartsPreviewComponent implements OnInit, OnChanges {
     @Input() hearingParts: HearingPartViewModel[];
     @Input() sessions: SessionViewModel[];
     @Output() selectHearingPart = new EventEmitter();
+    @ViewChild(MatSort) sort: MatSort;
+    @ViewChild(MatPaginator) paginator: MatPaginator;
 
     selectedHearingPart;
 
-    hearingPartsDataSource: MatTableDataSource<HearingPartViewModel>;
+    dataSource: MatTableDataSource<HearingPartViewModel>;
     displayedColumns = [
-      'case number',
-      'case title',
-      'case type',
-      'hearing type',
+      'selectHearing',
+      'caseNumber',
+      'caseTitle',
+      'caseType',
+      'hearingType',
       'duration',
-      'target schedule from',
-      'target schedule to',
-      'listed',
-      'select hearing',
-      'communication facilitator',
+      'communicationFacilitator',
       'priority',
-      'reserved judge',
-      'notes'
+      'reservedJudge',
+      'notes',
+      'scheduleStart',
+      'scheduleEnd',
     ];
 
     constructor(public dialog: MatDialog) {
         this.selectedHearingPart = new SelectionModel<HearingPartViewModel>(false, []);
     }
 
     ngOnInit() {
+        this.dataSource = new MatTableDataSource(Object.values(this.hearingParts));
     }
 
     ngOnChanges() {
-        this.hearingPartsDataSource = new MatTableDataSource(Object.values(this.hearingParts));
+        this.dataSource = new MatTableDataSource(Object.values(this.hearingParts));
+
+        this.dataSource.sortingDataAccessor = (item, property) => {
+            switch (property) {
+                case 'duration':
+                    return moment.duration(item[property]).asMilliseconds();
+
+                case 'reservedJudge':
+                    return item[property] === undefined ? null : item[property].name;
+
+                case 'priority':
+                    switch (item[property]) {
```

Comment:
After checking few things it looks, TS enums are not what I would wish for, as such could you put this switch statement as an exported function in the same file as Priority enum declaration and use it here?